include/lbann/io/persist.hpp

```diff
@@ -145,6 +209,62 @@ bool read_double (int fd, const char *name, double *val);
 bool write_string(int fd, const char *name, const char *buf, size_t size);
 bool read_string(int fd, const char *name, char *buf, size_t size);
 
+template <typename C>
+bool write_cereal_archive(C& obj, persist& p, persist_type pt, std::string suffix) {
```

Comment:
Why do these all return `bool`? It's probably better to just throw exceptions if things fail?